content/articles/implementation-of-dependency-injection-in-csharp/index.md

```diff
@@ -0,0 +1,387 @@
+
+Service classes 1 and 2 implement the Iclass interface, including the serve() method. The injector client class uses this method to inject services 1 and 2.
+
+The client class includes the `start()` method to set the object of the type class. The dependency is passed during the creation of the client object.
+
+### Implementing dependency injection using property injection
+Setter injection adds a dependency on a client class's public property. No new object creation or update is required.
+
+The dependency can also be passed through:
+- Using a setter method that stores the dependency in a variable.
+- Making use of lazy loading - That is, not initializing the concrete class until the property of the dependent class is called.
+
+The following is a c-sharp code implementing property injection:
+
+```c#
+using System;
+using System.Text;
+using System.Linq;
+using System.Collections.Generic;
+
+// Get the dependency relationship for the service element.
+public interface IClass{
+ // Initialize the method dependecy for class resource1 and resource2.
+ void Serve();
+}
+// Initialize serviceA
+public class ServiceA : IClass{
+ public void Serve() {
+    Console.WriteLine("ServiceA is Called");
+ }
+}
+// Initialize serviceB
+public class ServiceB : IClass{
+ public void Serve() {
+    Console.WriteLine("ServiceB is Called");
+ }
+}
+public class Client {
+ private IClass _service;
+ public IClass Service {
+ set {
+    _service = value;
+ }
+ }
+ public void Serve() {
+     _service.Serve();
+ }
+}
+class Program{
+ static void Main(string[] args){
+ Client client = new Client();
+
+ ServiceA sA = new ServiceA();
+ client.Service = sA;
+ client.Serve();
+
+ ServiceB sB = new ServiceB();
+ client.Service = sB;
+ client.Serve();
+ }
+}
+```
+
+Output:
+
+```bash
+ServiceA is Called
+serviceB is Called
+```
+
+From the above code, we have:
+
+```c#
+public class Client {
+ private IClass _service;
+ public IClass Service {
+ set { _service = value;}
+ }
+ public void Serve() {_service.Serve();}
+}
+```
+
+Client class is dependent on the property of type IClient through the `Iclass _service`.
+The set method sets the object property where the dependency is injected.
+
+We inject the dependency via a client class property. In this case, we will utilize a setter method. The dependency is then passed during client object creation.
+
+Property dependency injection is used when we intend to create the dependency object when it is required.
+
+### Implementing dependency injection using interface-based injections
+Interface-based injection injects a dependency into a client via an interface. The dependency becomes the injector.
+
+An interface-based injection is similar to property injection, using default getter and setter methods. However, interface-based injection sets the interface property using an explicit setter-getter interface.
+
+The following is a c-sharp code implementing property injection:
+
+```c#
```

Comment:
This code under interface-based injection doesn't seem to run on my end, kindly check for any build errors.